picard/script.py

```diff
@@ -910,6 +910,143 @@ def func_is_video(parser):
         return ""
 
 
+def func_find_str(parser, haystack, needle):
+    """Find the location of the first occurrence of one string within another.
+
+    Arguments:
+        haystack {str} -- The string to search
+        needle {str} -- The substring to find
+
+    Returns:
+        str -- Returns the zero-based index of the first occurrance of needle in haystack, or -1 if needle was not found.
+    """
+    return str(haystack.find(needle))
+
+
+def func_reverse_str(parser, text):
+    """Returns 'text' in reverse order.
+
+    Arguments:
+        text {str} -- Text to be processed
+
+    Returns:
+        str -- Text in reverse order.
+    """
+    return text[::-1]
+
+
+def func_substr(parser, text, start_index, end_index):
+    """Extract a specified portion of a string.
+
+    Arguments:
+        text {str} -- The string from which the extract will be made.
+        start_index {int} -- Index of the first character to extract.
+        end_index {int} -- Index of the first character that will not be extracted.
+
+    Returns:
+        str -- Returns the substring beginning with the character at the start index,
+               up to (but not including) the character at the end index.  The first
+               character is at index number 0.  If the start index is left blank, it
+               defaults to the first character in the string.  If the end index is
+               left blank, it defaults to the number of characters in the string.
+               If either index is negative, it is subtracted from the total number of
+               characters in the string to provide the index used.
+    """
+    _start = int(start_index) if start_index else None
+    _end = int(end_index) if end_index else None
+    return text[_start:_end]
+
+
+def func_get_multi_item(parser, multi, item_index, separator=MULTI_VALUED_JOINER):
+    """Returns value of the item at the specified index in the multi-value variable.  Index values are zero-based.
+
+    Arguments:
+        multi {multi-value} -- The multi-value from which the item is to be retrieved.
+        item_index {int} -- The zero-based index of the item to be retrieved.
+
+    Returns:
+        str -- Returns value of the item at the specified index in the multi-value variable.  Index values are zero-based.
+    """
+    if item_index:
+        _index = int(item_index.eval(parser))
+    else:
+        return ''
+    _multi = _get_multi_values(parser, multi, separator)
+    try:
+        return str(_multi[_index])
+    except IndexError:
+        return ''
+
+
+def func_foreach_multi_item(parser, multi, loop_code, _loop_value="_loop_value", _loop_count="_loop_count", separator=MULTI_VALUED_JOINER):
```

Comment:
Wrap arguments?

Reply:
I removed the option for different loop tag names (because I couldn't think of a case where alternate loop tag names would be necessary), and wrapped the rest.